go-controller/pkg/crd/userdefinednetwork/v1/shared.go

```diff
@@ -245,3 +245,30 @@ type IP string
 // +kubebuilder:validation:MaxItems=2
 // +kubebuilder:validation:XValidation:rule="size(self) != 2 || !isIP(self[0]) || !isIP(self[1]) || ip(self[0]).family() != ip(self[1]).family()", message="When 2 IPs are set, they must be from different IP families"
 type DualStackIPs []IP
+
+type TransportOption string
+type SNATOption string
+type RoutingOption string
+
+const (
+	TransportOptionNoOverlay TransportOption = "NoOverlay"
+	TransportOptionGeneve    TransportOption = "Geneve"
+
+	SNATEnabled  SNATOption = "Enabled"
+	SNATDisabled SNATOption = "Disabled"
+
+	RoutingManaged   RoutingOption = "Managed"
+	RoutingUnmanaged RoutingOption = "Unmanaged"
+)
+
+// NoOverlayOptions contains configuration options for networks operating in no-overlay mode.
+type NoOverlayOptions struct {
+	// OutboundSNAT defines the SNAT behavior for outbound traffic from pods.
+	// +kubebuilder:validation:Enum=Enabled;Disabled
+	// +required
+	OutboundSNAT SNATOption `json:"outboundSNAT"`
+	// Routing specifies whether the pod network routing is managed by OVN-Kubernetes or users.
```

Comment:
hmm the reason I made pod as workload is to account for VMs which are also run as pods in the end .. but here sounds like pod network is more easily decipherable, so feel free to drop my suggestion above for changing it to workloads unless you agree with that angle of keeping things generic